lib/dalli/protocol/base.rb

```diff
@@ -32,15 +32,24 @@ def request(opkey, *args)
         verify_state(opkey)
 
         begin
-          send(opkey, *args)
+          @connection_manager.start_request!
+          response = send(opkey, *args)
+
+          # pipelined_get emit query but doesn't read the response(s)
+          unless opkey == :pipelined_get
```

Comment:
I don't like having to single out that one `opkey`, but `request` is clearly defined as a choke point, so I assume going through another method isn't desirable as it may break various monitoring patches.

Also since it takes `*args`, I can't really have the caller pass an argument to tell the method that the request is incomplete.

But if you are ok with having a specialized `send_request` method or something like that, I'm happy to refactor this.